src/functions/module_lookup.ts

```diff
@@ -845,3 +847,120 @@ export const PIVOT = {
     return result;
   },
 } satisfies AddFunctionDescription;
+
+//--------------------------------------------------------------------------
+// OFFSET
+//--------------------------------------------------------------------------
+
+export const OFFSET = {
+  description: _t(
+    "Returns a range reference shifted by a specified number of rows and columns from a starting cell reference."
+  ),
+  args: [
+    arg(
+      "cell_reference (meta)",
+      _t("The starting point from which to count the offset rows and columns.")
+    ),
+    arg("offset_rows (number)", _t("The number of rows to offset by.")),
+    arg("offset_columns (number)", _t("The number of columns to offset by.")),
+    arg(
+      "height (number, default='height of cell_reference')",
+      _t("The number of rows of the range to return starting at the offset target.")
+    ),
+    arg(
+      "width (number, default='width of cell_reference')",
+      _t("The number of columns of the range to return starting at the offset target.")
+    ),
+  ],
+  compute: function (
+    cellReference: Maybe<{ value: string }>,
+    offsetRows: Maybe<FPayload>,
+    offsetColumns: Maybe<FPayload>,
+    height: Maybe<FPayload>,
+    width: Maybe<FPayload>
+  ) {
+    if (isEvaluationError(cellReference?.value)) {
+      return cellReference;
+    }
+
+    const _cellReference = cellReference?.value;
+    const zone = toZone(_cellReference!);
+
+    let offsetHeight = zone.bottom - zone.top + 1;
+    let offsetWidth = zone.right - zone.left + 1;
+
+    if (height) {
+      const _height = toNumber(height, this.locale);
+      assertPositive(
+        _t("Height value is %(_height)s. It should be greater than or equal to 1.", { _height }),
+        _height
+      );
+      offsetHeight = _height;
+    }
+
+    if (width) {
+      const _width = toNumber(width, this.locale);
+      assertPositive(
+        _t("width value is %(_width)s. It should be greater than or equal to 1.", { _width }),
```

Comment:
```suggestion
        _t("Width value is %(_width)s. It should be greater than or equal to 1.", { _width }),
```
Be consistent with the error message of `height`